hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -358,6 +380,48 @@ private void announce(HollowConsumer.ReadState readState) {
         }
     }
 
+    private class WriteEngineWrapper {
```

Comment:
I'd like to understand more about why the WriteEngineWrapper is necessary -- if we can do without it and if not, whether there is a better name to express its purpose